### PR TITLE
add identification features trough rever proxy remote headers

### PR DESCRIPTION
**Pull request by gallak**
Actually kanboard could authenticate trough reverse proxy headers but it couldn't fetch real name or mail without getting LDAP server or build the email like <loginname>@<remote domaine>.

This situation couldn't be applied in SAML context.

this PR permit to get mail and real username trough headers . Those headers are field by an SSO system like shibbolethized reverse proxy.  Old system are maintained.

I hope I'll break anything

### CODE REVIEW

**Comment by gallak** [comment]
you're right, I never use phpunit I'd to learn
thx

**Comment by gallak** [comment]
Hope the PR respect more code guidelines
Cheers

File: app/User/ReverseProxyUserProvider.php [hunk 1]
```diff
@@ -134,7 +140,12 @@ public function getName()
      */
     public function getEmail()
     {
-        return REVERSE_PROXY_DEFAULT_DOMAIN !== '' ? $this->username.'@'.REVERSE_PROXY_DEFAULT_DOMAIN : '';
+	if (! REVERSE_PROXY_MAIL_HEADER ){
+	        return REVERSE_PROXY_DEFAULT_DOMAIN !== '' ? $this->username.'@'.REVERSE_PROXY_DEFAULT_DOMAIN : '';
+	}else{
```
Comment (fguillot) [inline app/User/ReverseProxyUserProvider.php]: That would be appreciated if you could indent your code the same way as the rest to be more consistent.

File: app/constants.php [hunk 2]
```diff
@@ -7,154 +7,157 @@
...
-defined('LDAP_USER_CREATION') or define('LDAP_USER_CREATION', getenv('LDAP_USER_CREATION') ? strtolower(getenv('LDAP_USER_CREATION')) === 'true' : true);
-
-defined('LDAP_GROUP_ADMIN_DN') or define('LDAP_GROUP_ADMIN_DN', getenv('LDAP_GROUP_ADMIN_DN') ?: '');
-defined('LDAP_GROUP_MANAGER_DN') or define('LDAP_GROUP_MANAGER_DN', getenv('LDAP_GROUP_MANAGER_DN') ?: '');
-
-defined('LDAP_GROUP_PROVIDER') or define('LDAP_GROUP_PROVIDER', strtolower(getenv('LDAP_GROUP_PROVIDER')) === 'true');
-defined('LDAP_GROUP_BASE_DN') or define('LDAP_GROUP_BASE_DN', getenv('LDAP_GROUP_BASE_DN') ?: '');
-defined('LDAP_GROUP_FILTER') or define('LDAP_GROUP_FILTER', getenv('LDAP_GROUP_FILTER') ?: '');
-defined('LDAP_GROUP_USER_FILTER') or define('LDAP_GROUP_USER_FILTER', getenv('LDAP_GROUP_USER_FILTER') ?: '');
-defined('LDAP_GROUP_ATTRIBUTE_NAME') or define('LDAP_GROUP_ATTRIBUTE_NAME', getenv('LDAP_GROUP_ATTRIBUTE_NAME') ?: 'cn');
```
Comment (fguillot) [inline app/constants.php]: There is something wrong here. Your branch is not up to date with master.

**Comment by gallak** [comment]
code updated and sync with last upstream

**Comment by fguillot** [comment]
Please, rebase your PR and address the comments. Lot of PRs have been merged and that conflict with your changes.

Sorry for the delay.

**Pull request closed without merging**